### PR TITLE
A pull request on Awesome Books

### PR DESCRIPTION
**Pull request by dlwub**
We did the following things to implement the Awesome Books page:
- We created a collection that keeps a list of books
- We created a function to add a new book to the collection, with title and author, a function to remove a book from the collection, and a function to display all books saved in the collection in the top part of the page.
- When a user clicks the "Add" button, a new book is added to the collection and the new book is displayed on the page.
- When a user clicks the "Remove" button, the correct book is removed from the collection and the book disappears from the page.
- The data is preserved in the browser's memory by using and retrieved when the page reloads.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  addBookToArray(title, author);
+
+  // store data to localStorage to retain data on pageload
+  populateLocalStorage('bookList', booksArray);
+
+  // new book  to page
+  // addToPage(title, author)
+  bookList.innerHTML = '';
+  pageLoadChecks();
+});
```
Comment (vechicin) [inline index.js:96]: [OPTIONAL] Consider emptying the input fields after form submission, it'll make for better UI/UX.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    }
+  }
+}
+
+function populateLocalStorage(key, value) {
+  localStorage.setItem(key, JSON.stringify(value));
+}
+
+function removeBook(title) {
+  // booksArray = booksArray.filter((book) => book.title !== title);
```
Comment (vechicin) [inline index.js:72]: Please remove this commented line to keep a cleaner codebase.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  const title = titleForm.value;
+  const author = authorForm.value;
+
+  addBookToArray(title, author);
+
+  // store data to localStorage to retain data on pageload
+  populateLocalStorage('bookList', booksArray);
+
+  // new book  to page
+  // addToPage(title, author)
```
Comment (vechicin) [inline index.js:93]: Please remove this commented line to keep a cleaner codebase.

File: index.html [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      <header>
+        <h1>Awesome Books</h1>
+      </header>
+      <main>
+        <div id="books-list">
+        </div>
+
+        <form id="book-form" action="/" method="post">
+          <input id="title-input" type="text" placeholder="Title" name="title"><br/><br/>
+          <input id="author-input" type="text" placeholder="Auuthor" name="author"><br/><br/>
```
Comment (vechicin) [inline index.html:24]: [OPTIONAL] Consider changing the placeholder to `Author` to make for better UI/UX.

File: index.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    if (testResult[0]) {
+      // title in we can update author
+      booksArray[testResult[1]].Author = author;
+    } else {
+      const book = {
+        Title: title,
+        Author: author,
+      };
+      // not in
+      booksArray.push(book);
```
Comment (vechicin) [inline index.js:35]: [OPTIONAL] Consider using methods that don't mutate the array.

Screen Shot 2022-08-09 at 12 44 05

File: index.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+function populateLocalStorage(key, value) {
+  localStorage.setItem(key, JSON.stringify(value));
+}
+
+function removeBook(title) {
+  // booksArray = booksArray.filter((book) => book.title !== title);
+  const loc = inArray(title);
+  if (loc[0]) {
+    bookList.innerHTML = '';
+    booksArray.splice(loc[1], 1);
```
Comment (vechicin) [inline index.js:76]: [OPTIONAL] Consider using methods that don't mutate the array.

Screen Shot 2022-08-09 at 12 45 05

File: index.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    const testResult = inArray(title);
+
+    if (testResult[0]) {
+      // title in we can update author
+      booksArray[testResult[1]].Author = author;
+    } else {
+      const book = {
+        Title: title,
+        Author: author,
+      };
```
Comment (vechicin) [inline index.js:33]: [OPTIONAL] Consider giving books a unique `id`.

Screen Shot 2022-08-09 at 12 46 10

**Review by vechicin — CHANGES_REQUESTED** [review]
# Hi @dlwub and @kusiLaw ✋, 

Good job so far! 🔥 
There are some issues that you still need to work on to go to the next project but you are almost there! 🚨 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights ⚡ 

- [x] Good job implementing all of the features required for this milestone!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 📂 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @dlwub ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️ 
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

## Optional suggestions

- [ ] Kindly make sure your readme file is well formatted. Please remove the second author and make sure the correct Github handle is added.

##### _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

##### Cheers and Happy coding!:clap::clap::clap:
##### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**